src/main/java/org/jboss/tools/intellij/openshift/actions/component/DebugComponentAction.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.intellij.openshift.actions.component;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.ui.Messages;
+import org.jboss.tools.intellij.openshift.actions.OdoAction;
+import org.jboss.tools.intellij.openshift.tree.LazyMutableTreeNode;
+import org.jboss.tools.intellij.openshift.tree.application.ApplicationNode;
+import org.jboss.tools.intellij.openshift.tree.application.ComponentNode;
+import org.jboss.tools.intellij.openshift.utils.UIHelper;
+import org.jboss.tools.intellij.openshift.utils.odo.Component;
+import org.jboss.tools.intellij.openshift.utils.odo.Odo;
+
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+public class DebugComponentAction extends OdoAction {
+    public DebugComponentAction() {
+        super(ComponentNode.class);
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Odo odo) {
+        ComponentNode componentNode = (ComponentNode) selected;
+        Component component = (Component) componentNode.getUserObject();
+        ApplicationNode applicationNode = (ApplicationNode) componentNode.getParent();
+        LazyMutableTreeNode projectNode = (LazyMutableTreeNode) applicationNode.getParent();
+        CompletableFuture.runAsync(() -> {
```

Comment:
Don't use CompletableFuture.runAsync as it may block IJ are threads are limited. I remember there is any equivalent method in some help class

Reply:
if not using it, the IJ instance is freezing and becomes totally unusable. WDYT ?

Reply:
I was not claiming for removal but rather replacement by `ExecHelper.submit`